enhancements/security/security-profiles-operator.md

```diff
@@ -0,0 +1,629 @@
+is to leverage [Udica](https://github.com/containers/udica) to generate
+the profile.
+
+### SCC integration
+
+Security Context Constraints allow a user to restrict the security settings
+that a workload may have. Given that the profile settings are in the pod's
+`securityContext` it is imperative that we appropriately integrate with SCCs.
+
+Currently, the only SCC that allows users to set the Seccomp and SELinux
+settings for a pod is the privileged one. It looks as follows:
+
+```bash
+Name:                                           privileged
+Priority:                                       <none>
+Access:                                         
+  Users:                                        system:admin,system:serviceaccount:openshift-infra:build-controller
+  Groups:                                       system:cluster-admins,system:nodes,system:masters
+Settings:                                       
+  Allow Privileged:                             true
+  Allow Privilege Escalation:                   true
+  Default Add Capabilities:                     <none>
+  Required Drop Capabilities:                   <none>
+  Allowed Capabilities:                         *
+  Allowed Seccomp Profiles:                     *
+  Allowed Volume Types:                         *
+  Allowed Flexvolumes:                          <all>
+  Allowed Unsafe Sysctls:                       *
+  Forbidden Sysctls:                            <none>
+  Allow Host Network:                           true
+  Allow Host Ports:                             true
+  Allow Host PID:                               true
+  Allow Host IPC:                               true
+  Read Only Root Filesystem:                    false
+  Run As User Strategy: RunAsAny                
+    UID:                                        <none>
+    UID Range Min:                              <none>
+    UID Range Max:                              <none>
+  SELinux Context Strategy: RunAsAny            
+    User:                                       <none>
+    Role:                                       <none>
+    Type:                                       <none>
+    Level:                                      <none>
+  FSGroup Strategy: RunAsAny                    
+    Ranges:                                     <none>
+  Supplemental Groups Strategy: RunAsAny        
+    Ranges:                                     <none>
+```
+
+This basicallly allows for setting all security settings for a pod, which
+might not be what a user intends to allow.
+
+To enable integration with the SPO and promote better security in workloads,
+we could add another SCC called `measured` that would look similar to this:
+
+```bash
+Name:                                           measured
+Priority:                                       <none>
+Access:                                         
+  Users:                                        TBD
+  Groups:                                       TBD
```

Comment:
none?

Reply:
TBD == To Be Done. We need to give some thought to this.